packages/mobile/src/components/scrubber/Slider.tsx

```diff
@@ -0,0 +1,364 @@
+  e: GestureResponderEvent
+) => {
+  Animated.event(
+    [
+      null,
+      {
+        dx: animation
+      }
+    ],
+    { useNativeDriver: false }
+  )(e, { dx: newValue })
+}
+
+type SliderProps = {
+  /**
+   * A unique key to represent this instances of playback.
+   * If the user replays the same track, mediaKey should change
+   */
+  mediaKey: string
+  /**
+   * Whether audio is playing and the scrubber should animate
+   */
+  isPlaying: boolean
+  /**
+   * The duration of the currently playing track
+   */
+  duration: number
+  /**
+   * Callback invoked when focus is gained on the scrubber
+   */
+  onPressIn: () => void
+  /**
+   * Callback invoked when focus is lost on the scrubber
+   */
+  onPressOut: (percentComplete: number) => void
+  /**
+   * Callback invoked when dragging on the scrubber. A drag
+   * begins by grabbing the handle or pressing the rail
+   * and then dragging.
+   */
+  onDrag: (percentComplete: number) => void
+}
+
+/**
+ * Slider is an internal component of Scrubber (without the timestamps).
+ * It is separated as a different component so that it can be memoized
+ * independently in such a way where dragging the Slider does not cause
+ * re-renders of itself, despite having to update external timestamps on "drag."
+ */
+export const Slider = memo(
+  ({
+    mediaKey,
+    isPlaying,
+    duration,
+    onPressIn,
+    onPressOut,
+    onDrag
+  }: SliderProps) => {
+    const styles = useThemedStyles(createStyles)
+    const { primaryLight2, primaryDark2 } = useThemeColors()
+
+    // Animation to translate the handle and tracker
+    const translationAnim = useRef(new Animated.Value(0)).current
+    // Scale animation for the handle
+    const {
+      scale: handleScale,
+      handlePressIn: handlePressHandleIn,
+      handlePressOut: handlePressHandleOut
+    } = usePressScaleAnimation(HANDLE_GROW_SCALE)
+
+    const railRef = useRef<View>()
+    // The rail component's width
+    const [railWidth, setRailWidth] = useState(0)
+    // The rail component's distance from the left
+    const [railPageX, setRailPageX] = useState(0)
+    // The position of the drag-handle
+    const [handlePosition, setHandlePosition] = useState(0)
+
+    useEffect(() => {
+      if (railRef.current) {
+        railRef.current.measure((x, y, width, height, pageX, pageY) => {
+          setRailPageX(pageX)
+        })
+      }
+    }, [railRef])
+
+    const currentAnimation = useRef<Animated.CompositeAnimation>()
+    const play = useCallback(
+      (timeRemaining: number) => {
+        console.log(' play remaining ', timeRemaining)
+        currentAnimation.current = Animated.timing(translationAnim, {
+          toValue: railWidth,
+          duration: timeRemaining,
+          useNativeDriver: true
+        })
+        currentAnimation.current.start()
+      },
+      [translationAnim, railWidth]
+    )
+
+    const pause = useCallback(() => {
```

Comment:
onPause and onPlay maybe?

Reply:
This isn't a responder though -- it's the actual verb. Calling this fn actually triggers the "playing" or "pausing" of the animation. Wdyt?

Reply:
ah word yeah down with it!